packages/node-core/src/indexer/storeCache/storeCache.service.ts

```diff
@@ -95,15 +130,23 @@ export class StoreCacheService implements BeforeApplicationShutdown {
       const blockHeight = flushAll ? undefined : await this.metadata.find('lastProcessedHeight');
       // Get models that have data to flush
       const updatableModels = Object.values(this.cachedModels).filter((m) => m.isFlushable);
-
-      await Promise.all(updatableModels.map((model) => model.flush(tx, blockHeight)));
-
+      if (this._storeFlushInOrder) {
+        // 1. Independent(no associations) models can flush simultaneously
+        await Promise.all(
+          updatableModels.filter((m) => !m.hasAssociations).map((model) => model.flush(tx, blockHeight))
+        );
+        // 2. Models with associations will flush in orders,
+        await this.flushRelationalModelsInOrder(updatableModels, tx);
+      } else {
+        await Promise.all(updatableModels.map((model) => model.flush(tx, blockHeight)));
+      }
       await tx.commit();
     } catch (e) {
       logger.error(e, 'Database transaction failed');
       await tx.rollback();
       throw e;
     }
+    this.resetStoreOperationIndex();
```

Comment:
I don't think this will work. Remember that we take a snapshot (`updatableModels`) that we flush, so in the mean time items can still be added to the cache. That would mean that the storeOperationIndex could keep incrementing while we're flushing. 

We could store another number as to what index we have flushed up to.

I don't think we need to reset the index, its very unlikely there would be enough operations that we reach the limit of `number` and in that case we could just throw an error to restart the node and reset the index.